src/bytes.rs

```diff
@@ -417,6 +417,27 @@ impl Bytes {
         }
     }
 
+    /// Creates a new `Bytes` with the specified capacity, using stable storage
+    ///
+    /// Exactly like `with_capacity`, but never uses the inline optimization.
+    /// The resulting `Bytes` can be used with asynchronous C APIs that require
+    /// stable storage.
+    ///
+    /// # Examples
+    /// ```
+    /// use bytes::Bytes;
+    ///
+    /// let mut bytes = Bytes::with_capacity_stable(4);
+    ///
+    /// assert!(!bytes.is_inline());
+    /// ```
+    #[inline]
+    pub fn with_capacity_stable(capacity: usize) -> Bytes {
```

Comment:
This capability is already available by converting a vec. So, you would do: `Bytes::from(Vec::with_capacity(8))`. Do you think adding this constructor is still necessary?

Either way, the documentation should specifically call out that converting a Vec -> Bytes will keep the vec's storage intact.

Reply:
I suppose that would work since the specification does guarantee that `Vec` will always use heap storage [1].  But it feels hacky to tell consumers to use such a roundabout means of construction.  Ultimately it's up to you of course.  Let me know if you'd like me to change it.

[1] https://doc.rust-lang.org/std/vec/struct.Vec.html#guarantees

Reply:
I personally tend to err on the side of adding less new APIs. New APIs can always be added, but removing them is trickier.

If it turns out to be a point of confusion, we can add add the fn at that point.